### ISSUE TITLE
Update tests that wait on waitgroups to have timeout

### ISSUE THREAD

**Issue by tleyden** [couchbase/sync_gateway#3270]
As reported in https://github.com/couchbase/sync_gateway/issues/3268, the test essentially blocked forever after not getting expected results.

Rather than blocking forever, the test should instead fail with a timeout.  This is a separate issue from the underlying cause of https://github.com/couchbase/sync_gateway/issues/3268, and affects more tests... so I'm spinning out a separate issue / PR for this
